x/btcstaking/keeper/power_dist_change.go

```diff
@@ -181,6 +187,16 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 		for j := range dc.FinalityProviders[i].BtcDels {
 			btcDel := *dc.FinalityProviders[i].BtcDels[j]
 			if _, ok := unbondedBTCDels[btcDel.StakingTxHash]; !ok {
+				// Delegation is not unbonded (i.e., it's active)
+				if _, slashed := slashedBTCDels[btcDel.StakingTxHash]; slashed {
+					// If the BTC delegation is slashed, set its voting power to 0.
+					// This ensures that for this specific finality provider,
+					// the voting power of this delegation does not contribute
+					// to the total voting power. By modifying btcDel directly,
+					// we ensure that in subsequent heights, when the old cache
+					// is fetched, this delegation remains discounted for this FP.
+					btcDel.VotingPower = 0
+				}
 				fp.AddBTCDelDistInfo(&btcDel)
```

Comment:
- How about unrolling these two if statements? e.g., 
  - if unbonded then `continue`, 
  - if slashed then `continue`,
  - otherwise, execute `fp.AddBTCDelDistInfo(&btcDel)`